### PR TITLE
Fix for working on windows

### PR DESCRIPTION
**Pull request by hyrious**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ const childProcessArgs = () => {
 export const esbuildDev = async (options: Options) => {
   const workspaceRoot = findWorkspaceRoot(process.cwd()) || process.cwd();
   const workDir = await fs.mkdtemp(path.join(os.tmpdir(), "esbuild-dev"));
-  const syncSocketPath = path.join(workDir, "ipc.sock");
+  const syncSocketPath = path.join("\\\\?\\pipe", workDir, "ipc.sock");
```
Comment (airhorns) [inline src/index.ts:130]: My fault for not having tests yet, but does this still work on *nixes? I've never seen this syntax before!

File: src/Compiler.ts [hunk 2]
```diff
@@ -98,9 +98,16 @@ export class Compiler {
     const { tsConfig, tsConfigFile } = await this.getTSConfig(filename);
     if (this.tsConfigMap[tsConfigFile]) return;
 
+    let fileNames: string[];
+    if (process.platform === "win32") {
+      fileNames = tsConfig.fileNames.map(e => e.replace(/\//g, "\\"));
```
Comment (airhorns) [inline src/Compiler.ts:103]: Strange that TypeScript doesn't hand them out with the right separator.

**Review by airhorns — COMMENTED** [review]
Thanks for the PR!

File: src/index.ts (hunk 1 above)
Comment (hyrious) [reply src/index.ts:130]: Oh that's my fault, I will test & fix it on *nix (in fact MacOS) now.

File: src/Compiler.ts (hunk 2 above)
Comment (hyrious) [reply src/Compiler.ts:103]: It looks like TypeScript always hands out filenames with `/` separator.

**Pull request merged**